### ISSUE TITLE
Regression: Matched argument name and name of exported function breaks module export

### ISSUE THREAD

**Issue by tsofist** [Microsoft/TypeScript#11204]
**TypeScript Version:**  2.0.3

**Code**

```ts
export {
    my
}

function my() { }

function doSome1(my: any) {
    my = +my;
    return my;
}

function doSome2() {
    const internal = (my: any) => {
        my = +my;
        return my;
    };
    return internal("1");
}

doSome1("1");
doSome2();
```

**Expected behavior:**

```js
    function my() { }
    exports.my = my;
    function doSome1(my) {
        my = +my;
        return my;
    }
    function doSome2() {
        var internal = function (my) {
            my = +my;
            return my;
        };
        return internal("1");
    }
    doSome1("1");
    doSome2();
```

**Actual behavior:** 
```js
    function my() { }
    exports.my = my;
    function doSome1(my) {
        exports.my = my = +my;
        return my;
    }
    function doSome2() {
        var internal = function (my) {
            exports.my = my = +my;
            return my;
        };
        return internal("1");
    }
    doSome1("1");
    doSome2();
```

***Problem here:***
```js
exports.my = my = +my;
```

**Comment by RyanCavanaugh** [comment]
Is this a regression from 1.8?

**Comment by tsofist** [comment]
@RyanCavanaugh yes, it works fine on **1.8.10**

**Comment by tsofist** [comment]
Note that it work well if use
```ts
export function my() {}
```
instead of
```ts
export {
    my
}
```

**Comment by mhegazy** [comment]
This is the intended behavior. this was a change in the emitted code in TS 2.0 to match the ES6 module semantics. `export function my` exports a declaration `my`, changing it does not have any impact on that. whereas `export { my} ` exports the local binding for `my`. any change to the name should be witnessed by the importing module. see https://github.com/Microsoft/TypeScript/pull/8739 for more details.

**Comment by tsofist** [comment]
@mhegazy with all due respect, you are sure that #8739 is related to this situation described?
_Argument_ ```my``` of function ```doSome1``` has ***no relation*** to the export module

**Comment by mhegazy** [comment]
I actually did not see the function argument. my apologies. i was talking about the function name.

**Comment by mhegazy** [comment]
@vladima can you take a look. we need to fix this for release-2.0.5
